### ISSUE TITLE
Rotary dimmer Model 5121.10 not described in Zigbee-Adapter

### ISSUE THREAD

**Issue by berliaca** [ioBroker/ioBroker.zigbee#1866]
Greetings,

I need help with a zigbee device I bought. It is a rotary dimmer (Drehdimmer in german) of the company iluminize. The model number is 5121.10.

I was able to find the device in my zigbee network but I got the message that this device is not described in statesMapping. The device shows up in my dashboard of the zigbee adapter with a question mark as a thumbnail, but without any control element, such as on/off and the dimming level. Also in the object list the necessary adjustable parameters are missing.

My Zigbee adapter version is 1.8.23.

By searching the Internet, I was directed here, where you can report devices that are not described.

Is there any hope to make this device available for the IO Broker? 
I am attaching a few images that shows information about the device from the IO Broker.

Thanks in advance for your help.

Best Regards,
Alexander

![01](https://github.com/ioBroker/ioBroker.zigbee/assets/144383366/ed471ba3-0a22-4b1b-a960-0edad906f15a)
![02](https://github.com/ioBroker/ioBroker.zigbee/assets/144383366/82bd5491-18a8-4dbb-8de4-79262dd6b92b)
![04](https://github.com/ioBroker/ioBroker.zigbee/assets/144383366/7de1cd48-5939-4660-960f-e3c43fccca6d)
![07](https://github.com/ioBroker/ioBroker.zigbee/assets/144383366/4589e9a3-be03-4018-9d2c-9886a44a6373)

**Comment by gasinkas** [comment]
Having the same issue with this device. On shop website is written that it's compatible with Amazon Echo Plus (Alexa), IKEA TRÅDFRI, Osram Lightify, Google Assistant, uvm. So it should comply with zigbee standard... 

More information: https://www.iluminize.com/shop/product/5121-drehdimmer-mit-integriertem-zigbee-3-0-dimm-aktor-250w-500w-phasenabschnitt-fur-230v-leuchtmittel-180?category=90#attr=1224

**Comment by berliaca** [comment]
Hi, I fixed it by myself by just copy the code from another rotary dimmer model and changed the model number.

In the file /opt/iobroker/node_modules/zigbee-herdsman-converters/devices/iluminize.js I added

  {
         zigbeeModel: ['5121.10'],
         model: '5121.10',
         vendor: 'Iluminize',
         description: 'Rotary dimmer with integrated Zigbee 3.0 dimming actuator 230V',
         extend: extend_1.default.light_onoff_brightness({ noConfigure: true }),
         configure: async (device, coordinatorEndpoint, logger) => {
             await extend_1.default.light_onoff_brightness().configure(device, coordinatorEndpoint, logger);
             const endpoint = device.getEndpoint(1);
             await reporting.bind(endpoint, coordinatorEndpoint, ['genOnOff', 'genLevelCtrl']);
             await reporting.onOff(endpoint);
             await reporting.brightness(endpoint);
         },
   },

and it works now.

I am not familiar with the procedure here at github with the pull requests and so on. Perhaps a more experienced user could do this?

I also prepared a picture, since I could not link it to the device. 

![5121 10](https://github.com/ioBroker/ioBroker.zigbee/assets/144383366/ad7b83f6-d1ba-4951-9189-89b204b0ca5d)

Best Regards,
Alexander

**Comment by gasinkas** [comment]
Amazing!! Keep in mind that you have to do it everytime you update your zigbee adapter. I forgot doing it after updating the adapter 🙈 
Where do I copy the picture?

**Comment by berliaca** [comment]
Yeah, I backed up my modified file for this case.
I found the folders with all the other pictures of the zigbee devices, but i cannot link it to the rotary dimmer.
I converted the picture to PNG and named it first 5121.10.png and then iluminize_5121_10.png (and illuminize_5121_10.png) like the other iluminize devices. 

But it did not work. I started the zigbee adapter several times afterwards.

The file paths are
/opt/iobroker/iobroker-data/files/zigbee.admin/img/
/opt/iobroker/node_modules/iobroker.zigbee/admin/img/

Maybe you have another idea?
